### PR TITLE
Feat/orderCount by timeframe

### PR DESCRIPTION
**Pull request by eketeUg**
you can check this one that was used for the previous cohort

# Fixes Issue/Linear Ticket
https://linear.app/zuri-project-backend/issue/COM-26/api-for-getting-orders-count-in-a-timeframe

> This could be an existing issue or a linear ticket
>
> - Github Issue Example: My PR Closes #{ISSUE}
> - Linear Ticket Example: Fixes ID-#{ISSUE}

# Changes proposed

> Talk about the things you did eg. files changes, dependencies installed e.t.c
Added endpoint  with query params  to get the count of orders made with a timeframe like: today, yesterday, one-week-ago and two-weeks-ago, this will help in analytical purposes

# Check List (Check all the applicable boxes)

:rotating_light:Please review the [style guide for contributing](add link here) and [guidelines for contributing](add link here) to this repository.

- [x] My code follows the code style of this project.
- [x] This PR does not contain plagiarized content.
- [x] The title and description of the PR is clear and explains the approach.
- [x] I am making a pull request against the **main branch** (left side).
- [x] My commit messages styles matches our requested structure.
- [x] My code additions will fail neither code linting checks nor unit test.
- [x] I am only making changes to files I was requested to.

# Screenshots/ Videos

Screenshot testing with "today" query: https://drive.google.com/file/d/1GbSV1AasEYsqEsUbEyXKhq14J7g32npx/view?usp=drivesdk

Screenshot testing with "yesterday" query :https://drive.google.com/file/d/1PaAHAf72GzTOqvvVMg3a4aaisjzM7ASO/view?usp=drivesdk

I also implemented others timeframes like a week and two weeks ago: https://drive.google.com/file/d/1R3ylI62CaDTkAArs3CuieRVYwzWDhQgi/view?usp=drivesdk

 https://drive.google.com/file/d/1LO-edpFCeui6gacfo5SGobVEUXp_0AUB/view?usp=drivesdk

### CODE REVIEW

**Comment by Benrobo** [comment]
resolve conflict

**Comment by Benrobo** [comment]
resolve your conflict

**Comment by eketeUg** [comment]
Boss I have resolved the conflicts

**Comment by eketeUg** [comment]
The more conflicts keeps coming up after merging other PRs

**Review by AirZED — APPROVED** [review]

File: src/controller/discount.controller.ts [hunk 1]
```diff
@@ -1,159 +1,159 @@
...
+//     this.success(res, '--discount/success', `Successfully created discount`, 201, {
+//       promo: promo,
+//       productWithPromo: createdProdDiscount ?? null,
+//     });
+//   }
+
+//   async trackDiscount(req: Request, res: Response) {
+//     const userId = (req as any).user?.id ?? TestUserId;
+//     // const userId = 'dcb5b46a-9391-474c-9e69-fe37cfe821e9';
+//     const validateSchema = createDiscountSchema.validate(req.body);
```
Comment (Benrobo) [inline src/controller/discount.controller.ts:139]: Why would you comment out a code someone worked on and add your yarn-error.log? why!!! .. do you wanna break the feature someone has been implementing

**Review by Benrobo — CHANGES_REQUESTED** [review]
your code can't be accepted.. 

pull and update with the dev branch

**Comment by eketeUg** [comment]
> your code can't be accepted..
> 
> pull and update with the dev branch

Done .

**Pull request merged**